control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -165,6 +166,17 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 	}
 
+	// Determine if the expected control plane operator image matches the active image in the control-plane-operator leader.
+	// If it doesn't need to no-op until the operator is recreated and at the matching version
+	if _, ok := hostedControlPlane.Annotations[hyperv1.ControlPlaneOperatorImageAnnotation]; !ok {
+		return ctrl.Result{}, fmt.Errorf("expected controlPlaneOperatorImageAnnotation not present on HCP")
+	}
+	expectedImage := hyperv1.ControlPlaneOperatorImageAnnotation
```

Comment:
Not clear to me how this is working — `expectedImage` is being assigned to a constant... shouldn't this value come from the map lookup on L171? Seems like this is a mistake and is always `false`?

Reply:
it should be from the map you are correct